core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -410,36 +409,29 @@ public VersionNumber getRequiredVersion() {
         }
     }
 
-    private static final List<DetachedPlugin> DETACHED_LIST = Collections.unmodifiableList(Arrays.asList(
-            new DetachedPlugin("maven-plugin", "1.296", "1.296"),
-            new DetachedPlugin("subversion", "1.310", "1.0"),
-            new DetachedPlugin("cvs", "1.340", "0.1"),
-            new DetachedPlugin("ant", "1.430.*", "1.0"),
-            new DetachedPlugin("javadoc", "1.430.*", "1.0"),
-            new DetachedPlugin("external-monitor-job", "1.467.*", "1.0"),
-            new DetachedPlugin("ldap", "1.467.*", "1.0"),
-            new DetachedPlugin("pam-auth", "1.467.*", "1.0"),
-            new DetachedPlugin("mailer", "1.493.*", "1.2"),
-            new DetachedPlugin("matrix-auth", "1.535.*", "1.0.2"),
-            new DetachedPlugin("windows-slaves", "1.547.*", "1.0"),
-            new DetachedPlugin("antisamy-markup-formatter", "1.553.*", "1.0"),
-            new DetachedPlugin("matrix-project", "1.561.*", "1.0"),
-            new DetachedPlugin("junit", "1.577.*", "1.0"),
-            new DetachedPlugin("bouncycastle-api", "2.16.*", "2.16.0"),
-            new DetachedPlugin("command-launcher", "2.86.*", "1.0")
-    ));
+    /** Record of which plugins which removed from core and when. */
+    private static final List<DetachedPlugin> DETACHED_LIST;
 
     /** Implicit dependencies that are known to be unnecessary and which must be cut out to prevent a dependency cycle among bundled plugins. */
-    private static final Set<String> BREAK_CYCLES = new HashSet<String>(Arrays.asList(
-            "script-security/matrix-auth",
-            "script-security/windows-slaves",
-            "script-security/antisamy-markup-formatter",
-            "script-security/matrix-project",
-            "script-security/bouncycastle-api",
-            "script-security/command-launcher",
-            "credentials/matrix-auth",
-            "credentials/windows-slaves"
-    ));
+    private static final Set<String> BREAK_CYCLES;
+
+    static {
+        List<DetachedPlugin> detached = new ArrayList<>();
+        try (InputStream is = ClassicPluginStrategy.class.getResourceAsStream("/jenkins/split-plugins.txt")) {
+            for (String line : org.apache.commons.io.IOUtils.readLines(is, StandardCharsets.UTF_8)) {
+                String[] pieces = line.split(" ");
```

Comment:
I would propose to skip lines starting from `#` and then to document the format directly in the resource files. It would help future contributors

Reply:
Possible. Would need to make a matching change in the downstream PR, and then restart the manual test cycle.

Reply:
Comments would be really useful in order to reference plugin split tickets and other info. I am fine if we create a follow-up JIRA ticket and proceed with this change without such patches

Reply:
I am not sure it could be done safely as a follow-up: again, the downstream PR needs to match the format.

Thirty seconds to write the code for such a change, an hour to recreate the test environment needed to verify it. I will see if I can find time for it, but I do not know.